doc/programming_model/alignment.md

```diff
@@ -0,0 +1,26 @@
+Memory Alignment Requirements {#dev_guide_alignment}
```

Comment:
(nit) Having whole section for memory alignment looks like an overkill. Maybe we could add this info to existing section like [Input Validation Notes](https://uxlfoundation.github.io/oneDNN/dev_guide_c_and_cpp_apis.html#input-validation-notes).

+@ranukund for additional insights.

Reply:
Thanks, moved this to the API page and put a second-level section. Please take a look.

Reply:
LGTM

Reply:
Agree that this is rather obscure detail that does not belong in library introduction.